lib/streamlit/runtime/pages_manager.py

```diff
@@ -227,18 +45,23 @@ def __init__(
     ):
         self._main_script_path = main_script_path
         self._main_script_hash: PageHash = calc_md5(main_script_path)
-        self.pages_strategy = PagesManager.DefaultStrategy(self, **kwargs)
         self._script_cache = script_cache
         self._intended_page_script_hash: PageHash | None = None
         self._intended_page_name: PageName | None = None
         self._current_page_script_hash: PageHash = ""
+        self._pages: dict[PageHash, PageInfo] | None = None
         # A relic of v1 of Multipage apps, we performed special handling
         # for apps with a pages directory. We will keep this flag around
         # for now to maintain the behavior for apps that were created with
         # the pages directory feature.
-        PagesManager.uses_pages_directory = Path(
-            self.main_script_parent / "pages"
-        ).exists()
+        #
+        # NOTE: we will update the feature if the flag has not been set
+        #       this means that if users use v2 behavior, the flag will
+        #       always be set to False
```

Comment:
is the reason for this that PagesManagerV2 is always instantiated before strategy v1 could be instantiated?

Reply:
I'm not sure if I understand but the flow is:

1. Instantiation of PagesManager detects the `pages` folder and sets the setting
2. Script Runner detects this behavior and performs the v1 logic
3. script _actually_ uses `st.navigation`, it resets the the setting to always be using V2.
 
The note basically says that this can be updated in other places, and why. Let me know if you think there's a clearer way to describe this.

Reply:
Thank you! 

With regards to
> it resets the the setting to always be using V2

where does this reset happen? Is a new `PagesManager` instantiated in `script_runner`? I am not seeing it right now, so probably I am looking at the wrong place

Reply:
It's reset here. https://github.com/streamlit/streamlit/blob/refactor/remove-mpa-v1/lib/streamlit/commands/navigation.py#L219

Since it's a class variable it will be shared across instances.